File: Algorithms/Happy Number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        hashtable={}
        while(n!=1):
            temp=0
            while(n>0):
                temp+=(n%10)**2
                n//=10
            if temp in hashtable:
                return False
            hashtable[temp]=None
            n=temp
        return True

File: Algorithms(Java)/Happy Number.java
class Solution {
    public boolean isHappy(int n) {
        HashSet<Integer> h=new HashSet<Integer>();
        while(n!=1){
            if (n<=3)
                return false;
            else{
                if (h.contains(n))
                    return false;
                h.add(n);
                int temp=0;
                while(n>0){
                    temp+=Math.pow((n%10),2);
                    n=n/10;
                }
                n=temp;
            }
        }
        return true;
    }
}